.github/workflows/image-push.yml

```diff
@@ -0,0 +1,46 @@
+name: Build And Push
+on:
+  push:
+    tags:
+      - 'v*'
+jobs:
+  push-multiarch:
+    name: Build and Push amd64 and arm64
+    runs-on: ubuntu-latest
+    env:
+      TAG: ${{github.ref_name}}
+      REPOSITORY_OWNER: ${{ github.repository_owner }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{github.ref}}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Docker Registry Login
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
```

Comment:
Since EIO have successfully added secrets to repos https://github.com/rancherlabs/eio/issues/2106#issuecomment-2037119595, you should be using the eio action to mount the vault secrets